source/Engine.h

```diff
@@ -102,6 +102,19 @@ class Engine {
 
 
 private:
+	class DoubleBuffer {
+	public:
+		constexpr operator size_t() const { return val; }
+
+		constexpr DoubleBuffer() : val(0) {}
+
+		inline void next() { val = !val; }
+
+	private:
+		size_t val;
+
+	};
```

Comment:
I don't really feel like this class name works to "self document" the code and make it more clear what the purpose of the two variables that are being changed are. I think it'd be much simpler to just add a comment above the variables in the header describing what their purpose is and what true and false actually represent for the arrays that they're being used on.

Perhaps also leave a comment above the DrawList, BatchDrawList, and Radar arrays describe why they are arrays. For someone who knows their stuff about graphics, they might be able easily tell that they're buffers, but no harm in leaving a comment that explains that. I think tandem with the comment above the frame variables, that'd serve the same purpose as the DoubleBuffer while being more clear.

Reply:
Is the new description satisfactory?